### ISSUE TITLE
build failure with rustc nightly

### ISSUE THREAD

**Issue by matthiaskrgr** [redox-os/orbclient#39]
````
rustc 1.24.0-nightly (4a7c072fa 2017-12-25)
cargo 0.25.0-nightly (e08f31018 2017-12-24)
````
````
cargo build
    Updating registry `https://github.com/rust-lang/crates.io-index`
   Compiling num-traits v0.1.41                                                 
   Compiling libc v0.2.34
   Compiling lazy_static v0.2.11
   Compiling bitflags v0.7.0
   Compiling sdl2-sys v0.30.0
   Compiling rand v0.3.18
   Compiling num-integer v0.1.35
   Compiling num-iter v0.1.34
   Compiling num v0.1.41
   Compiling sdl2 v0.30.0
   Compiling orbclient v0.3.12 (file:///tmp/orbclient)
error: borrow of packed field requires unsafe function or block (error E0133)
  --> src/color.rs:59:31
   |
59 |         write!(f, "{:#010X}", self.data)
   |                               ^^^^^^^^^
   |
note: lint level defined here
  --> src/lib.rs:9:9
   |
9  | #![deny(warnings)]
   |         ^^^^^^^^
   = note: #[deny(safe_packed_borrows)] implied by #[deny(warnings)]
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #46043 <https://github.com/rust-lang/rust/issues/46043>

error: borrow of packed field requires unsafe function or block (error E0133)
  --> src/renderer.rs:59:27
   |
59 |                 let old = &mut data[y as usize * w as usize + x as usize].data;
   |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #46043 <https://github.com/rust-lang/rust/issues/46043>

error: aborting due to 2 previous errors

error: Could not compile `orbclient`.

To learn more, run the command again with --verbose.
````

**Comment by Miniwoffer** [comment]
Removing "repr(packed)" in Color.rs seems to resolve this(i only tried the exsample on linux ).
But im guessing that isnt a true fix, it is there for a reason after all.

**Comment by robbycerantola** [comment]
fixed with PR #43
